### PR TITLE
install_provider.sh

### PR DESCRIPTION
**Pull request by etam**
[APPS-148](https://golemproject.atlassian.net/browse/APPS-148)
Erigon provider node installation script (nginx, yagna, golemsp, erigon binaries & exe-unit's runtime).

~There is one TODO to be resolved, once installation of yagna itself is implemented.~

### Followin env variables are used during installation
| name | default value | description |
| :-------- | ----------------: | :------------- |
| AAA | aaa | WIP todo; |

### How to test / run installation script
```bash
sudo env \
   ERIGON_HOSTNAME=X.erigon.golem.network \
   ERIGON_EMAIL=admin@golem.network \
  bash install_provider.sh
```

### CODE REVIEW

**Comment by etam** [comment]
file to configure access: /etc/nginx/erigon_htpasswd

https://nginx.org/en/docs/http/ngx_http_auth_basic_module.html#auth_basic_user_file

**Review by Wiezzel — APPROVED** [review]
Looks OK, but I'm no expert in bash.
If it worked on providers, it's fine.

**Comment by etam** [comment]
I haven't tested this yet on providers. I tested it locally on a VM with `ERIGON_DISABLE_SSL=y`. But it's close enough.

File: install_provider.sh [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+nginx -t || die "nginx doesn't like it's config"
+systemctl restart nginx
+systemctl enable --now reload-nginx.path
+
+# ========= yagna =========
+
+# install erigon runtime before yagna itself
+mkdir -p "${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon"
+cat >"${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon.json" <<EOF
```
Comment (pnowosie) [inline install_provider.sh:234]: There are multiple files generated like this and for the services I think it's OK. I think runtime related files we'd be more happy to have them outside of the script included separately. I can imagine that regarding our handbook effort we'd like to point some file and explain something inside it... WDYT @etam ?

I think we can prepare a folder inside a repo containing all runtime files and `curl` them in the script. Later we can simplify link (e.g. point at master or other branch - but it's important right now...

File: install_provider.sh [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+]
+EOF
+curl -sSfL https://github.com/golemfactory/yagna-service-erigon/releases/download/3deaadc/ya-erigon-runtime.tar.gz \
+    | tar -xzf - -C "${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon" \
+    || die "failed to install erigon runtime"
+
+mkdir -p "${ERIGON_DATADIR}/goerli"
+mkdir -p "${ERIGON_USER_HOME}/.local/share/ya-erigon-runtime"
+
+cat >"${ERIGON_USER_HOME}/.local/share/ya-erigon-runtime/ya-erigon-runtime.json" <<EOF
```
Comment (pnowosie) [inline install_provider.sh:253]: This file get out of the script from repo? (see prev comment)

File: install_provider.sh [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    "${ERIGON_USER_HOME}/.local/lib/yagna" \
+    "${ERIGON_USER_HOME}/.local/share/ya-erigon-runtime"
+
+# install yagna
+curl -sSf https://join.golem.network/as-provider \
+    | sudo -i -u "$ERIGON_USER" -- env YA_INSTALLER_CORE=pre-rel-v0.7.0-rc6 bash -
+# for some reason even on success installer exits with code 1
+
+sudo -i -u golem ya-provider preset deactivate wasmtime
+sudo -i -u golem ya-provider preset deactivate vm
```
Comment (pnowosie) [inline install_provider.sh:274]: Hmm 🤔 - but how we activate the `erigon` runtime ❓ 
It went missed or it's kind of golemsp black magic?

File: install_provider.sh [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+Type=oneshot
+ExecStart=/usr/bin/systemctl reload nginx
+
+[Install]
+WantedBy=default.target
+EOF
+
+systemctl daemon-reload
+
+cat >/etc/nginx/nginx.conf <<EOF
```
Comment (pnowosie) [inline install_provider.sh:112]: We might benefit of bringing this file outside the script as well (just for the readability and ease of further modification)

**Review by pnowosie — CHANGES_REQUESTED** [review]
I think we could bring some file out of the script (see comments)

File: install_provider.sh (hunk 3 above)
Comment (etam) [reply install_provider.sh:274]: It's black magic, although I'm not sure if it's in installer, golemsp, or ya-provider itself. On the first run all runtimes are enabled and installer sets presets for all of them.

File: install_provider.sh (hunk 4 above)
Comment (etam) [reply install_provider.sh:112]: I can move it out of the script, but I'm using here some bash variables substitution magic. I'd have to change it into some `sed` magic or something.

File: install_provider.sh (hunk 2 above)
Comment (etam) [reply install_provider.sh:253]: I don't think it's justified. It's short, contains a lot of site-specific config and almost no common boilerplate.

File: install_provider.sh (hunk 4 above)
Comment (pnowosie) [reply install_provider.sh:112]: Ok, lets not complicate it further - it seems to be easiest approach.

**Review by pnowosie — APPROVED** [review]
Please DO NOT merge it yet.
New version of the runtime is just after the corner, I will then upload it using this script and correct possible issues.

**Comment by mateuszsrebrny** [comment]
APPS-148 # connecting PR to jira

**Pull request merged**